snuba/datasets/storages/processors/replaced_groups.py

```diff
@@ -49,7 +49,11 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
             if final:
                 metrics.increment(
                     "final",
-                    tags={"cause": "final_flag", "referrer": request_settings.referrer},
+                    tags={
+                        "cause": "final_flag",
+                        "referrer": request_settings.referrer,
```

Comment:
I have the same concern expressed in detail below. Referrer should not have been a request setting. See below for details